pypeit/core/wavecal/wv_fitting.py

```diff
@@ -178,20 +282,24 @@ def iterative_fitting(spec, tcent, ifit, IDs, llist, disp,
             flg_continue = False
         # Fit with rejection
         xfit, yfit, wfit = tcent[ifit], all_ids[ifit], weights[ifit]
-        mask, fit = utils.robust_polyfit(xfit/xnspecmin1, yfit, n_order, function=func, sigma=sigrej_first,
-                                         minx=fmin, maxx=fmax, verbose=verbose, weights=wfit)
+        #mask, fit = utils.robust_polyfit(xfit/xnspecmin1, yfit, n_order, function=func, sigma=sigrej_first,
+        #                                 minx=fmin, maxx=fmax, verbose=verbose, weights=wfit)
+        pypeitFit = fitting.robust_fit(xfit/xnspecmin1, yfit, n_order, function=func,
+                                       lower=sigrej_first, upper=sigrej_first,
+                                       minx=fmin, maxx=fmax, weights=wfit)
 
-        rms_ang = utils.calc_fit_rms(xfit[mask == 0]/xnspecmin1, yfit[mask == 0], fit, func, minx=fmin, maxx=fmax,
-                                     weights=wfit[mask == 0])
+        rms_ang = pypeitFit.calc_fit_rms()#xfit[pypeitFit.gpm == 0]/xnspecmin1,
+                                     #yfit[pypeitFit.gpm == 0],
+                                     #weights=wfit[pypeitFit.gpm == 0])
         rms_pix = rms_ang/disp
         if verbose:
             msgs.info('n_order = {:d}'.format(n_order) + ': RMS = {:g}'.format(rms_pix))
 
         # Reject but keep originals (until final fit)
-        ifit = list(ifit[mask == 0]) + sv_ifit
+        ifit = list(ifit[pypeitFit.gpm == 0]) + sv_ifit
```

Comment:
I believe this is a bug.The mask convention in robust_polyfit is not documented, but looking at the code mask is a bad pixel mask, i.e. True = bad. Whereas for robust_polyfit_djs, the mask is a good pixel mask, i.e. True=Good. So I believe this should be ifit[pypeFit.gpm] to grab the good elements.

Reply:
Fixed by me.